pydra/engine/core.py

```diff
@@ -546,8 +575,8 @@ def pickle_task(self):
         """ Pickling the tasks with full inputs"""
         pkl_files = self.cache_dir / "pkl_files"
         pkl_files.mkdir(exist_ok=True, parents=True)
-        task_main_path = pkl_files / f"{self.name}_{self.checksum}_task.pklz"
-        save(task_path=pkl_files, task=self, name_prefix=f"{self.name}_{self.checksum}")
+        task_main_path = pkl_files / f"{self.name}_{self.uid}_task.pklz"
+        save(task_path=pkl_files, task=self, name_prefix=f"{self.name}_{self.uid}")
```

Comment:
are you creating separate task files for each split. that may not be optimal in the previous implementation you had one file from which you read all the states right?

Reply:
yes, one file. what is your concern?

Reply:
if it's one file, i don't understand (at this point) why each state split has a different uid? where is that being used?

Reply:
let me think... i keep braking things today, so perhaps should double check this

Reply:
this is called for the task with full input, not for every state separately